### ISSUE TITLE
Tags Do Not Save to Posts Consistently

### ISSUE THREAD

**Issue by stamalden** [WordPress/gutenberg#9971]
**Describe the bug**
When added via the Gutenberg editor, Tags are not being saved to posts. However, they appear within the Tags list My Sites > Tags.

**To Reproduce**
1.	Open a new post in Gutenberg editor
2.	Click on "Tags” to expand the menu item
3.	Insert a new Tag and hit “enter”
4.	After hitting “enter”, the new Tag appears in greyscale with mirror text. Clicking enter again removes the mirror text. 
5.	Click the arrow on "Tags" menu to collapse the menu selection. Open the menu selection again and newly entered Tag disappears
6.	Publish the post and view the post on the front end and the Tag does not appear. If the Tag does appear, edit the post and Tag disappears again

**Expected behavior**
Tags should save consistently to posts and be visable on the front-end of the site. Similar issue noted for adding Categories.

**Screenshots**
https://snag.gy/BSa6Mj.jpg

**Desktop:**
 - Windows
 - Browser [e.g. chrome, IE, firefox]

**Comment by commorancy** [comment]
If it's a theme based problem, you can look at my site which is exhibiting the problem and determine if it is a theme problem. I can also make you a video of the behavior and post a link here if you like. Editing with Gutenberg on mine, I can't get tags or categories to stick at all. I have to use the older new editor to set these.

**Comment by Copons** [comment]
Pasting the same answer I've given on Automattic/wp-calypso#28974:

I can reproduce on Core Gutenberg (v4.6.0):

![nov-30-2018 14-41-28](https://user-images.githubusercontent.com/2070010/49295631-28c0a200-f4ae-11e8-8279-5cf3a78abb96.gif)

It seems to be a race condition kind of conflict caused by the fact that, every time you input a tag it first fires a `GET /tags?search=foobar` to see if it's an existing tag, and then when hitting enter `POST /tags { name: "foobar" }` to save the tag.

What seems to happen is that most of the `POST` requests end up as:

```json
{
  "code": "term_exists",
  "message": "A term with the name provided already exists in this taxonomy.",
  "data": {
    "status": 409,
    "term_id": 123
  }
}
```

What I can't seem to figure out, is why some tags get inserted anyway (e.g. in my tests I've always tried with `1 2 3 4 5` etc, so `1` definitely already existed when I recorded the gif, and yet it got added correctly anyway.
